### PR TITLE
fixes sentry errors

### PR DESCRIPTION
**Pull request by m-abs**
- crashed on android with used deleted object
- crashed on older android with missing media button handler

### CODE REVIEW

**Review by 838 — APPROVED** [review]
LGTM 👍

File: src/media-service.android.ts [hunk 1]
```diff
@@ -352,28 +352,67 @@ export namespace dk {
...
+           * state itself, even though the name sounds like it."
+           */
+          androidx.core.content.ContextCompat.startForegroundService(
+            this.getApplicationContext(),
+            new android.content.Intent(this, dk.nota.MediaService.class),
+          );
+          this.startForeground(notificationId, notification);
+          this._isForegroundService = true;
+
+          this.exoPlayer.setForegroundMode(true);
```
Comment (ddfreiling) [inline src/media-service.android.ts]: Not sure if we even need this, read docs on https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/SimpleExoPlayer.html#setForegroundMode-boolean-

File: src/media-service.android.ts [hunk 2]
```diff
@@ -352,28 +352,67 @@ export namespace dk {
...
 
-        /**
-         * This may look strange, but the documentation for [Service.startForeground]
-         * notes that "calling this method does *not* put the service in the started
-         * state itself, even though the name sounds like it."
-         */
-        androidx.core.content.ContextCompat.startForegroundService(this.getApplicationContext(), new android.content.Intent(this, dk.nota.MediaService.class));
-        this.startForeground(notificationId, notification);
-        this._isForegroundService = true;
+        if (shouldBeInForeground === this._isForegroundService) {
```
Comment (ddfreiling) [inline src/media-service.android.ts:373]: `if (!shouldBeInForeground || this._isForegroundService)` would have made more sense and make below check on `shouldBeInForeground` unnecessary.

File: src/media-service.android.ts [hunk 3]
```diff
@@ -303,85 +307,131 @@ export namespace dk {
...
-          this.exoPlayer.release();
-          this.exoPlayer = undefined;
-        }
+        delete this._mediaSessionMetadataProvider;
+
+        this._mediaSession?.setActive(false);
+        this._mediaSession?.release();
+        delete this._mediaSession;
+
+        this.exoPlayer?.setForegroundMode(false);
```
Comment (ddfreiling) [inline src/media-service.android.ts]: Already did that.

File: src/media-service.android.ts [hunk 4]
```diff
@@ -303,85 +307,131 @@ export namespace dk {
...
+
+          if (trace.isEnabled()) {
+            trace.write(`${this.cls}._handleNotificationPosted(${notificationId}, ${notification}) - started in foreground`, notaAudioCategory);
+          }
 
-        this.exoPlayer?.setForegroundMode(true);
+          return;
+        }
+
+        this.stopForeground(false);
```
Comment (ddfreiling) [inline src/media-service.android.ts]: Sending true as param here will remove the notification. Maybe we should do that on `STATE_ENDED` ?

File: src/platforms/android/include.gradle [hunk 5]
```diff
@@ -1,8 +1,10 @@
...
-  implementation 'com.google.android.exoplayer:exoplayer-core:2.11.+'
-  // implementation 'com.google.android.exoplayer:exoplayer-dash:2.10.+'
-  implementation 'com.google.android.exoplayer:exoplayer-ui:2.11.+'
-  implementation 'com.google.android.exoplayer:extension-cast:2.11.+'
-  implementation 'com.google.android.exoplayer:extension-mediasession:2.11.+'
+  def exoplayerVersion = "2.11.3+"
+
+  // implementation "com.google.android.exoplayer:exoplayer:$exoplayerVersion"
+  // implementation "com.google.android.exoplayer:exoplayer-dash:$exoplayerVersion"
+  // implementation "com.google.android.exoplayer:extension-cast:$exoplayerVersion"
```
Comment (ddfreiling) [inline src/platforms/android/include.gradle:6]: can remove comments

**Review by ddfreiling — CHANGES_REQUESTED** [review]
Nice work, some minor corrections & questions.

File: src/media-service.android.ts (hunk 2 above)
Comment (m-abs) [reply src/media-service.android.ts:373]: I'm not sure I understand.

Here we check if we already are in the wanted state, the check below is meant to start or stop the foreground service.

Comment (ddfreiling) [reply src/media-service.android.ts:373]: Ah, I didn’t read far enough down in the method. It’s good as it is

**Comment by ddfreiling** [comment]
LGTM now, did you agree that `setForegroundMode` was not necessary for audio playback?

**Comment by m-abs** [comment]
> LGTM now, did you agree that `setForegroundMode` was not necessary for audio playback?

The reference implementation in `android/uamp` doesn't use it, so I assume it is not needed.

**Pull request merged**